### PR TITLE
Remove isLightTheme flag from highlight API

### PR DESCRIPTION
**Pull request by eseliger**
After we migrated to CSS variables for syntax highlighting, we don't need this anymore! That means we won't reload code blobs after a theme change anymore - less flicker :)

### CODE REVIEW

File: client/web/src/integration/streaming-search-mocks.ts [hunk 1]
```diff
@@ -30,10 +30,9 @@ export const diffSearchStreamEvents: SearchEvent[] = [
...
...ourcegraphBaseUrl + &#39;/search?&#39; + urlQuery)\n</span></div></td></tr><tr><td class="line" data-line="6"></td><td class="code"><div><span style="color:#268bd2;">@@ -434,0 +439,4 @@ </span><span style="color:#cb4b16;">describe(&#39;Search regression test suite&#39;, () =&gt; {\n</span></div></td></tr><tr><td class="line" data-line="7"></td><td class="code"><div><span style="background-color:#deeade;color:#2b3750;">+        })\n</span></div></td></tr><tr><td class="line" data-line="8"></td><td class="code"><div><span style="background-color:#deeade;color:#2b3750;">+        test(&#39;Forked repos included by by fork option&#39;, async () =&gt; {\n</span></div></td></tr><tr><td class="line" data-line="9"></td><td class="code"><div><span style="background-color:#deeade;color:#2b3750;">+            const urlQuery = buildSearchURLQuery(&#39;type:repo sgtest/mux fork:yes&#39;, GQL.SearchPatternType.regexp, false)\n</span></div></td></tr><tr><td class="line" data-line="10"></td><td class="code"><div><span style="background-color:#deeade;color:#2b3750;">+            await driver.page.goto(config.sourcegraphBaseUrl + &#39;/search?&#39; + urlQuery)</span></div></td></tr></tbody></table>',
```
Comment (eseliger) [inline client/web/src/integration/streaming-search-mocks.ts:35]: I think these fixtures are outdated, but I don't know how to fix it

Comment (eseliger) [reply client/web/src/integration/streaming-search-mocks.ts:35]: @limitedmage maybe you have advice :)

File: client/web/src/enterprise/batches/preview/list/VisibleChangesetApplyPreviewNode.tsx [hunk 2]
```diff
@@ -45,9 +45,9 @@ export interface VisibleChangesetApplyPreviewNodeProps extends ThemeProps {
 
 export const VisibleChangesetApplyPreviewNode: React.FunctionComponent<VisibleChangesetApplyPreviewNodeProps> = ({
     node,
-    isLightTheme,
     history,
     location,
+    isLightTheme,
```
Comment (limitedmage) [inline client/web/src/enterprise/batches/preview/list/VisibleChangesetApplyPreviewNode.tsx:50]: This one is still used?

File: client/web/src/integration/streaming-search-mocks.ts (hunk 1 above)
Comment (limitedmage) [reply client/web/src/integration/streaming-search-mocks.ts:35]: I got it by copying response from a valid call to `highlightCode` and `HighlightedFile`, we should update these with the new class-based syntax highlighting responses

File: client/web/src/repo/blob/BlobPage.tsx [hunk 3]
```diff
@@ -333,9 +319,9 @@ export const BlobPage: React.FunctionComponent<Props> = props => {
                     settingsCascade={props.settingsCascade}
                     onHoverShown={props.onHoverShown}
                     history={props.history}
-                    isLightTheme={isLightTheme}
                     telemetryService={props.telemetryService}
                     location={props.location}
+                    isLightTheme={props.isLightTheme}
```
Comment (limitedmage) [inline client/web/src/repo/blob/BlobPage.tsx:324]: Is this one still needed?

File: client/web/src/repo/compare/RepositoryCompareDiffPage.tsx [hunk 4]
```diff
@@ -84,8 +76,8 @@ interface RepositoryCompareDiffPageProps
     extends RepositoryCompareAreaPageProps,
         RouteComponentProps<{}>,
         PlatformContextProps,
-        ExtensionsControllerProps,
-        ThemeProps {
+        ThemeProps,
```
Comment (limitedmage) [inline client/web/src/repo/compare/RepositoryCompareDiffPage.tsx:79]: Is this prop still needed?

**Review by limitedmage — APPROVED** [review]
🎉

**Review by camdencheek — APPROVED** [review]
Changes look good to me, but this is mostly a rubber stamp since I'm unfamiliar with the frontend code. 

Happy to help regenerate those test values with the new class format if needed

File: client/web/src/enterprise/batches/preview/list/VisibleChangesetApplyPreviewNode.tsx (hunk 2 above)
Comment (eseliger) [reply client/web/src/enterprise/batches/preview/list/VisibleChangesetApplyPreviewNode.tsx:50]: Yes the line decorations still require it

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 5]
```diff
@@ -1089,7 +1089,7 @@ type Query {
     """
     EXPERIMENTAL: Syntax highlights a code string.
     """
-    highlightCode(code: String!, fuzzyLanguage: String!, disableTimeout: Boolean!, isLightTheme: Boolean!): String!
+    highlightCode(code: String!, fuzzyLanguage: String!, disableTimeout: Boolean!): String!
```
[the remark this answers is not in the document]
Comment (eseliger) [reply cmd/frontend/graphqlbackend/schema.graphql:1092]: Hm yeah good idea. I'll make it optional for now and file a ticket to remove it in a future version.

**Comment by limitedmage** [comment]
From the Percy snapshots, looks like the search query ranges aren't being highlighted correctly? Is this being introduced by this PR or just outdated mock results?

**Comment by eseliger** [comment]
Oh good question. I just copied in an API result from dotcom, it seems to work there, so maybe I just copied the results wrong .. 🤔

**Comment by chrispine** [comment]
This one has been sitting here for a while. What's the state of it now?

**Comment by eseliger** [comment]
This diff is quite outdated. Superseded by this PR: https://github.com/sourcegraph/sourcegraph/pull/24671

**Pull request closed without merging**